src/lib.rs

```diff
@@ -1834,6 +1834,17 @@ impl Connection {
             })? as usize
         };
 
+        // Make sure the buffer is same or larger than an explicit
```

Comment:
Alternatively, I tried
```@@ -1824,7 +1824,13 @@ impl Connection {
         let payload_len = if hdr.ty == packet::Type::Short {
             b.cap()
         } else {
-            b.get_varint().map_err(|e| {
+            b.get_varint()
+                .and_then(|x| {
+                    if x > b.cap() as u64
+                    { Err(octets::BufferTooShortError) }
+                    else
+                    { Ok(x) } })
+                .map_err(|e| {
                 drop_pkt_on_err(
                     e.into(),
                     self.recv_count,
```
good thing is no need to double check for short packet but doesn't look much useful...